main.go

```diff
@@ -223,6 +223,11 @@ func handlePost(res http.ResponseWriter, req *http.Request) {
 
 	date, _ := time.ParseInLocation("2006-01-02", dateStr, loc)
 
+	if date.Weekday() != 4 && !time.IsZero() {
```

Comment:
Also, while you're on it, switch the conditionals around. First the validity-check, then the specific check :) And, strictly speaking, `date.IsZero()` would imply `date.Weekday() != 4`, but I think it's still good to have both checks for readability.